src/executor/alter_executor.cpp

```diff
@@ -0,0 +1,187 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// alter_executor.cpp
+//
+// Identification: src/executor/alter_executor.cpp
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "executor/alter_executor.h"
+
+#include "catalog/catalog.h"
+#include "catalog/table_catalog.h"
+#include "common/logger.h"
+#include "executor/executor_context.h"
+#include "storage/data_table.h"
+
+namespace peloton {
+namespace executor {
+
+// Constructor for alter table executor
+AlterExecutor::AlterExecutor(const planner::AbstractPlan *node,
+                             ExecutorContext *executor_context)
+    : AbstractExecutor(node, executor_context) {}
+
+// Initialize executor
+// Nothing to initialize for now
+bool AlterExecutor::DInit() {
+  LOG_TRACE("Initializing Alter Executer...");
+
+  LOG_TRACE("Alter Executor initialized!");
+  return true;
+}
+
+bool AlterExecutor::DExecute() {
+  LOG_TRACE("Executing Alter...");
+  bool result = false;
+  const planner::AlterPlan &node = GetPlanNode<planner::AlterPlan>();
+  auto current_txn = executor_context_->GetTransaction();
+  AlterType type = node.GetAlterTableType();
+  switch (type) {
+    case AlterType::RENAME:
+      result = RenameColumn(node, current_txn);
+      break;
+    case AlterType::ALTER:
+      result = AlterTable(node, current_txn);
+      break;
+    default:
+      throw NotImplementedException(StringUtil::Format(
+          "Alter Type not supported, %s", AlterTypeToString(type).c_str()));
+  }
+  return result;
+}
+
+bool AlterExecutor::RenameColumn(
+    const peloton::planner::AlterPlan &node,
+    peloton::concurrency::TransactionContext *txn) {
+  auto database_name = node.GetDatabaseName();
+  auto table_name = node.GetTableName();
+  auto schema_name = node.GetSchemaName();
+  auto new_column_name = node.GetNewName();
+  auto old_column_name = node.GetOldName();
+
+  ResultType result = catalog::Catalog::GetInstance()->RenameColumn(
```

Comment:
[race condition]
If there are two transactions trying to rename a column in a table at the same time, it is not safe to do operations on catalog instance without locks.

Reply:
We planed to add lock logic here and in the alterTable, should we merge your lock manager and executor implementation?

Reply:
Sure. Although our current implementation of lock manager is correct, we are still working on it to support more functions. Most interface may not change in future, but I suggest that you keep an eye on our future pull request to guarantee your codes can run on our latest version of lock manager and index executors.